src/components/CsvUpload/ValidateCsvTable.tsx

```diff
@@ -47,7 +52,20 @@ export function ValidateCsvTable<T extends { [K in keyof T]: string | number }>(
         cell: (info: CellContext<PreviewRow<T>, unknown>) => {
           const field = info.row.original[key]
 
-          let value: string | number = field.raw
+          // Handle null/undefined fields for optional columns
+          if (!field) {
+            return (
+              <span className={classNames(
+                'Layer__table-cell-content',
+                `${CSS_PREFIX}__cell-content`,
+              )}
+              >
+                {/* Empty cell for null values */}
+              </span>
+            )
+          }
+
+          let value: string | number | null | undefined = field.raw
           if (field.is_valid) {
```

Comment:
If value is falsy (undefined, null, etc.) then nothing will get rendered inside the cell anyway. This should work:
```suggestion
          let value: string | number | null | undefined = field.raw
          const isValid = field && field.is_valid
          if (isValid) {
```